plugin/src/test/java/com/microsoft/java/bs/gradle/plugin/Conversions.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package com.microsoft.java.bs.gradle.plugin;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+import com.microsoft.java.bs.gradle.model.impl.DefaultJavaExtension;
+
+/**
+ * Utils to convert objects between two different class loaders to avoid
+ * ClassCastException.
+ */
+public class Conversions {
+
+  /**
+   * Convert an object to a DefaultJavaExtension.
+   */
+  public static DefaultJavaExtension toJavaExtension(Object object) {
```

Comment:
Why not do this copy in `DefaultGradleSourceSet` constructor like the other classes are copied?  Seems messy to use reflection.

Reply:
If without this util. To get a java extension, for example, I need to write:

```java
DefaultJavaExtension javaExtension = (DefaultJavaExtension) gradleSourceSet.getExtensions().get("java");
```

Then when I run the unit tests, I got `ClassCastException` saying that the classes are loaded by different class loaders.

I haven't dig deep into this issue, the solution(workaround) I found is to use such util to do a conversion.